### PR TITLE
Easier access for vehicles' group window.

### PR DESCRIPTION
**Pull request by stormcone**
This is the follow-up PR of #7582.

I changed the shift+click combinations to control+click, removed the right click options from the vehicle news window, and added the functionality to the vehicle lists windows also.

I as well moved the button from the bottom of the vehicle news window to the caption bar, because I am not really liked the way it looked.
vehicle_news

### CODE REVIEW

**Comment by andythenorth** [comment]
I tested this briefly.  Shared order vehicle lists / station vehicle list are **remarkably improved** by the ctrl-click behaviour 😄 

I'm not sure about the icon on the order warning: I tried it, but I didn't have any opinion whether it's an improvement, nor did I test it for edge cases etc  I'll leave it to someone else to decide if that button is in scope or not, but the ctrl-click behaviour should definitely be merged IMHO.

**Comment by stormcone** [comment]
I mainly put the button on the vehicle news window for the old vehicle notifications. I usually replace the vehicles only when they getting old and can not be "autorenewed". So in that case easier to open the group window with the button.

File: src/group_gui.cpp [hunk 1]
```diff
@@ -369,6 +369,7 @@ class VehicleGroupWindow : public BaseVehicleListWindow {
 		this->groups.ForceRebuild();
 		this->groups.NeedResort();
 		this->BuildGroupList(vli.company);
+		this->group_sb->SetCount((uint)this->groups.size());
```
Comment (LordAro) [inline src/group_gui.cpp:372]: SetCount should probably actually take a uint, but currently it takes an int, so I'm not sure what this cast is doing

File: src/group_gui.cpp [hunk 2]
```diff
@@ -992,6 +997,29 @@ class VehicleGroupWindow : public BaseVehicleListWindow {
 	{
 		if (this->vehicle_sel == vehicle) ResetObjectToPlace();
 	}
+
+	void SelectGroup(const GroupID g_id) {
```
Comment (LordAro) [inline src/group_gui.cpp]: function opening brace on newline

Comment (LordAro) [reply src/group_gui.cpp]: also some doxygen comments

File: src/group_gui.cpp [hunk 3]
```diff
@@ -1013,18 +1041,30 @@ static WindowDesc _train_group_desc(
...
+		w = AllocateWindowDescFront<VehicleGroupWindow>(&_other_group_desc, num, need_existing_window);
 	}
+	if (w != nullptr) w->SelectGroup(group);
+}
+
+/**
+ * Show the group window for the given vehicle.
+ * @param v The vehicle to show the window for.
+ */
+void ShowCompanyGroupForVehicle(const Vehicle *v) {
```
Comment (LordAro) [inline src/group_gui.cpp]: function opening brace on newline

File: src/lang/english.txt [hunk 4]
```diff
@@ -870,6 +870,8 @@ STR_NEWS_NEW_VEHICLE_NOW_AVAILABLE                              :{BIG_FONT}{BLAC
 STR_NEWS_NEW_VEHICLE_TYPE                                       :{BIG_FONT}{BLACK}{ENGINE}
 STR_NEWS_NEW_VEHICLE_NOW_AVAILABLE_WITH_TYPE                    :{BLACK}New {STRING} now available!  -  {ENGINE}
 
+STR_NEWS_SHOW_VEHICLE_GROUP_TOOLTIP                             :{BLACK}Open the group window and select the vehicle's group
```
Comment (LordAro) [inline src/lang/english.txt]:
```suggestion
STR_NEWS_SHOW_VEHICLE_GROUP_TOOLTIP                             :{BLACK}Open the group window focused on the vehicle's group
```
?

File: src/news_gui.cpp [hunk 5]
```diff
@@ -279,6 +283,31 @@ struct NewsWindow : Window {
...
 
+		NWidgetCore *nwid = this->GetWidget<NWidgetCore>(WID_N_SHOW_GROUP);
+		if (ni->reftype1 == NR_VEHICLE && nwid != nullptr) {
+			const Vehicle *v = Vehicle::Get(ni->ref1);
+			switch (v->type) {
+				case VEH_TRAIN:
+					nwid->widget_data = STR_TRAIN;
+					break;
+				case VEH_ROAD:
+					if (RoadVehicle::From(v)->IsBus()) {
```
Comment (LordAro) [inline src/news_gui.cpp]: perhaps a ternary?

**Review by LordAro — CHANGES_REQUESTED** [review]
Nothing significant. Looks good!

File: src/group_gui.cpp (hunk 1 above)
Comment (stormcone) [reply src/group_gui.cpp:372]: IIRC I copied that line from here, so that's why it's there :)
https://github.com/OpenTTD/OpenTTD/blob/40ebf3c7659201b5365839b1371c4f9e84feea50/src/group_gui.cpp#L515

Comment (LordAro) [reply src/group_gui.cpp:372]: Weird. Don't worry about it then, it can be fixed later (when SetCount is actually converted to a uint)

**Review by LordAro — APPROVED** [review]
